### ISSUE TITLE
How to modify path resolution.

### ISSUE THREAD

**Issue by justinbmeyer** [less/less.js#2792]
Is it possible to modify less's path resolution without making a custom build?

I work on [StealJS](http://stealjs.com), which supports using AMD/CJS/import to load from npm packages (and has supported LESS [at least 5 years](https://github.com/bitovi/legacy-steal/tree/8a6aa5b9bcd2327b115e59a075d137b377bc39c3/less)).  In short, people want to be able to write:

```
/* ROOT/user.less */
@import "package/styles.less"
````

And `styles.less` might look like:

```
/* ROOT/node_modules/package/styles.less */
@import "bootstrap/less/mixins.less"
```

Both `package` and  `bootstrap` are be located somewhere within `node_modules` that I'm able to locate.

My guess is that other module loaders will like this same feature (or have already figured out how to do it).

I've already written a "pre-less" transpiler that takes:

```
/* ROOT/user.less */
@import "package/styles.less"
```

And converts it to something like:

```
/* ROOT/user.less */
@import "./node_modules/package/styles.less"
```

The problem is that I can't also run my "pre-less" transpiler on `styles.less`.  There's no hook for path resolution or when the file gets loaded.  So `styles.less` doesn't get updated with the right path to bootstrap:

```
/* ROOT/node_modules/package/styles.less */
@import "../../node_modules/bootstrap/mixins.less"
```

## TLDR;

Are there hooks for either:

- intercepting the less source code less is loading on its own.
- intercepting path resolution logic.

Thanks!

**Comment by matthew-dean** [comment]
@justinbmeyer There's a conversation thread on #2615 that you'd probably want to read through. By the end it had gotten close to consensus, other than the keywords. Maybe you could put in your $0.02.

**Comment by matthew-dean** [comment]
Specifically, @seven-phases-max objected to the word "modules" as one of the options. As neither of us are asking for the feature, maybe some suggestions could come for people who want this. Does this match your use case?

**Comment by seven-phases-max** [comment]
As I understand this is exactly what #2618. So merging this to #2618 (re-open if I'm wrong).
